feat(language_server): tell clients to watch for .oxlintrc.json files

Sysix opened the pull request:
Now it is the job of the server to watch / unwatch for files.
The watching will still do the client, but the logic is now under the server's control.

Changing configuration or workspaces still needs a restart because there needs to be an Identifier-Structure for all specific watcher. This will be done in the next weeks ;)

camc314 commented:
### Merge activity

* **May 18, 8:11 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(language_server): tell clients to watch for .oxlintrc.json files (#11078)

Now it is the job of the server to watch / unwatch for files.
The watching will still do the client, but the logic is now under the server's control.

Changing configuration or workspaces still needs a restart because there needs to be an Identifier-Structure for all specific watcher. This will be done in the next weeks ;)
